lang/java/compiler/src/main/velocity/org/apache/avro/compiler/specific/templates/java/classic/record.vm

```diff
@@ -129,6 +129,25 @@ public class ${this.mangle($schema.getName())}#if ($schema.isError()) extends or
   }
 #end
 
+#if ($this.hasLogicalTypeField($schema))
```

Comment:
I think you're right: this is inside the getter/setter loop on line 110. Thanks, @ppearcy! I'll post a new version with the fix.